### DISCUSSION TITLE
I can't install Lunar

### DISCUSSION THREAD

**Discussion by tarikkamat** [lunarphp/lunar#2036] [Q&A]
I use Laravel v11.35.0 (PHP v8.4.1)

`composer require lunarphp/lunar -W`

______

./composer.json has been updated
Running composer update lunarphp/lunar --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires lunarphp/lunar * -> satisfiable by lunarphp/lunar[0.1.1, ..., 0.8.1].
    - lunarphp/core[0.1.6, ..., 0.2.7] require laravel/framework ^8.0|^9.0 -> found laravel/framework[v8.0.0, ..., v8.83.29, v9.0.0, ..., v9.52.18] but it conflicts with your root composer.json require (^11.31).
    - lunarphp/core[0.3.0, ..., 0.8.1] require laravel/framework ^9.0|^10.0 -> found laravel/framework[v9.0.0, ..., v9.52.18, v10.0.0, ..., v10.48.25] but it conflicts with your root composer.json require (^11.31).
    - lunarphp/lunar[0.1.1, ..., 0.1.5] require lunarphp/licensing dev-main -> found lunarphp/licensing[dev-feature/laravel_10, 0.1, ..., 0.5.x-dev, 2.0-beta] but it does not match the constraint.
    - lunarphp/lunar 0.1.6 requires lunarphp/core 0.1.6 -> satisfiable by lunarphp/core[0.1.6].
    - lunarphp/lunar 0.2.0 requires lunarphp/core 0.2.0 -> satisfiable by lunarphp/core[0.2.0].
    - lunarphp/lunar 0.2.1 requires lunarphp/core 0.2.1 -> satisfiable by lunarphp/core[0.2.1].
    - lunarphp/lunar 0.2.2 requires lunarphp/core 0.2.2 -> satisfiable by lunarphp/core[0.2.2].
    - lunarphp/lunar 0.2.3 requires lunarphp/core 0.2.3 -> satisfiable by lunarphp/core[0.2.3].
    - lunarphp/lunar 0.2.4 requires lunarphp/core 0.2.4 -> satisfiable by lunarphp/core[0.2.4].
    - lunarphp/lunar 0.2.5 requires lunarphp/core 0.2.5 -> satisfiable by lunarphp/core[0.2.5].
    - lunarphp/lunar 0.2.6 requires lunarphp/core 0.2.6 -> satisfiable by lunarphp/core[0.2.6].
    - lunarphp/lunar 0.2.7 requires lunarphp/core 0.2.7 -> satisfiable by lunarphp/core[0.2.7].
    - lunarphp/lunar 0.3.0 requires lunarphp/core 0.3.0 -> satisfiable by lunarphp/core[0.3.0].
    - lunarphp/lunar 0.3.1 requires lunarphp/core 0.3.1 -> satisfiable by lunarphp/core[0.3.1].
    - lunarphp/lunar 0.3.2 requires lunarphp/core 0.3.2 -> satisfiable by lunarphp/core[0.3.2].
    - lunarphp/lunar 0.4.0 requires lunarphp/core 0.4.0 -> satisfiable by lunarphp/core[0.4.0].
    - lunarphp/lunar 0.4.1 requires lunarphp/core 0.4.1 -> satisfiable by lunarphp/core[0.4.1].
    - lunarphp/lunar 0.4.2 requires lunarphp/core 0.4.2 -> satisfiable by lunarphp/core[0.4.2].
    - lunarphp/lunar 0.5.0 requires lunarphp/core 0.5.0 -> satisfiable by lunarphp/core[0.5.0].
    - lunarphp/lunar 0.5.1 requires lunarphp/core 0.5.1 -> satisfiable by lunarphp/core[0.5.1].
    - lunarphp/lunar 0.5.2 requires lunarphp/core 0.5.2 -> satisfiable by lunarphp/core[0.5.2].
    - lunarphp/lunar 0.5.3 requires lunarphp/core 0.5.3 -> satisfiable by lunarphp/core[0.5.3].
    - lunarphp/lunar 0.5.4 requires lunarphp/core 0.5.4 -> satisfiable by lunarphp/core[0.5.4].
    - lunarphp/lunar 0.5.5 requires lunarphp/core 0.5.5 -> satisfiable by lunarphp/core[0.5.5].
    - lunarphp/lunar 0.5.6 requires lunarphp/core 0.5.6 -> satisfiable by lunarphp/core[0.5.6].
    - lunarphp/lunar 0.6.0 requires lunarphp/core 0.6.0 -> satisfiable by lunarphp/core[0.6.0].
    - lunarphp/lunar 0.6.1 requires lunarphp/core 0.6.1 -> satisfiable by lunarphp/core[0.6.1].
    - lunarphp/lunar 0.6.2 requires lunarphp/core 0.6.2 -> satisfiable by lunarphp/core[0.6.2].
    - lunarphp/lunar 0.6.3 requires lunarphp/core 0.6.3 -> satisfiable by lunarphp/core[0.6.3].
    - lunarphp/lunar 0.6.4 requires lunarphp/core 0.6.4 -> satisfiable by lunarphp/core[0.6.4].
    - lunarphp/lunar 0.7.0 requires lunarphp/core 0.7.0 -> satisfiable by lunarphp/core[0.7.0].
    - lunarphp/lunar 0.7.1 requires lunarphp/core 0.7.1 -> satisfiable by lunarphp/core[0.7.1].
    - lunarphp/lunar 0.7.2 requires lunarphp/core 0.7.2 -> satisfiable by lunarphp/core[0.7.2].
    - lunarphp/lunar 0.7.3 requires lunarphp/core 0.7.3 -> satisfiable by lunarphp/core[0.7.3].
    - lunarphp/lunar 0.7.4 requires lunarphp/core 0.7.4 -> satisfiable by lunarphp/core[0.7.4].
    - lunarphp/lunar 0.7.5 requires lunarphp/core 0.7.5 -> satisfiable by lunarphp/core[0.7.5].
    - lunarphp/lunar 0.7.6 requires lunarphp/core 0.7.6 -> satisfiable by lunarphp/core[0.7.6].
    - lunarphp/lunar 0.8.0 requires lunarphp/core 0.8.0 -> satisfiable by lunarphp/core[0.8.0].
    - lunarphp/lunar 0.8.1 requires lunarphp/core 0.8.1 -> satisfiable by lunarphp/core[0.8.1].

You can also try re-running composer require with an explicit version constraint, e.g. "composer require lunarphp/lunar:*" to figure out if any version is installable, or "composer require lunarphp/lunar:^2.1" if you know which you need.

_________

I try 

`composer require lunarphp/lunar:^2.1`

_____

./composer.json has been updated
Running composer update lunarphp/lunar
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires lunarphp/lunar ^2.1, found lunarphp/lunar[dev-main, dev-0.7_0.8_branch_merge, dev-1.0.0-alpha, dev-1.0.0-beta, 0.1.0-rc.1, ..., 0.x-dev, 1.0.0-alpha.1, ..., 1.x-dev] but it does not match the constraint.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

_______

**Comment by dualklip** [comment]
Did you changed the minimum stability to dev?

https://docs.lunarphp.io/core/installation.html#composer-require-package

**Comment by glennjacobs** [comment]
Did you update your composer.json to set "minimum-stability": "dev"?
